### DISCUSSION TITLE
Double reply to multicast pings from backbone network

### DISCUSSION THREAD

**Discussion by pjasinski990** [openthread/openthread#9153] [Q&A] [answered]
For some addresses I can see two multicast ICMP ping responses from the end device when pinging from backbone net.

I'm running OTBR revision `48dae8e2eca4f0fa90ad1a11cd5dfa21d9e33cfc` on a Raspberry PI 3B+ and my end device is an nRF5340DK board running nrf openthread CLI sample (sdk-nrf revision is tag `2.4.0`). 

Build steps (assuming nRF Connect SDK installed in current directory):
 * `cd ncs/nrf/samples/openthread/cli`
 * `west build -p -b nrf5340dk_nrf5340_cpuapp`
 * `west flash`

Network topology is border router as a leader and the nrf DK as a router.

Expected behaviour:
One icmp ping response received from the end device.

Observed behaviour:
For some addresses, the board responds twice:
```
PING ff05::5(ff05::5) 56 data bytes
64 bytes from 2a02:a31a:a045:3380:ba27:ebff:fe4e:3eb8: icmp_seq=1 ttl=64 time=
[double_reply.zip](https://github.com/openthread/openthread/files/11677030/double_reply.zip)
228 ms
64 bytes from fd11:22::dc81:8948:bab5:d4f2: icmp_seq=1 ttl=63 time=228 ms
64 bytes from fd11:22::dc81:8948:bab5:d4f2: icmp_seq=1 ttl=63 time=228 ms
```

Attached wireshark capture of the occurence.

**Comment by pjasinski990** [comment]
[double_reply.zip](https://github.com/openthread/openthread/files/11677035/double_reply.zip)

**Comment by jwhui** [comment, marked as answer]
The multiple replies could be due to having multiple IPv6 stacks handling the ICMPv6 Echo Request.

On POSIX platform where the host stack has its own IPv6 stack, we explicitly disable OpenThread's handling of ICMPv6 Echo Request messages:
https://github.com/openthread/openthread/blob/4c5320168c9dfe6cfb811d13239b18dfba41e7e2/src/posix/platform/netif.cpp#L1946
